ruby/bps-kafka/lib/bps/kafka.rb

```diff
@@ -4,34 +4,52 @@
 module BPS
   module Kafka
     # ReliablePublisher is a slow, but reliable Kafka Publisher.
-    # There is no batching at all, messages are emitted synchronously.
+    # There is no batching at all, messages are delivered one by one.
     class ReliablePublisher < BPS::Publisher::Abstract
-      DELIVER_MESSAGE_RETRIES = 3 # 1 by kafka-ruby's Kafka::Client default, increase slightly
-
       class Topic < BPS::Publisher::Topic::Abstract
-        def initialize(client, topic_name)
-          @client = client
-          @topic_name = topic_name
+        def initialize(producer, topic_name, opts = {})
```

Comment:
`**opts` pls -> ruby 2.7

Reply:
I'll remove `opts` in Topic constructor, no need.